File: error/src/main/java/yun/blog/error/error/ErrorCode.java
package yun.blog.error.error;

import lombok.Getter;

@Getter
public enum ErrorCode {

  // Common
  INVALID_INPUT_VALUE(400, "C001", "입력값이 올바르지 않습니다."),
  METHOD_NOT_ALLOWED(405, "C002", "올바른 요청이 아닙니다."),
  INTERNAL_SERVER_ERROR(500, "C004", "문제가 발생했습니다. "),

  ;


  private final String code;
  private final String message;
  private int status;

  ErrorCode(final int status, final String code, final String message) {
    this.status = status;
    this.message = message;
    this.code = code;
  }


}

File: msa-error-response/src/main/kotlin/com/example/msaerrorresponse/ErrorCode.kt
package com.example.msaerrorresponse

import com.fasterxml.jackson.annotation.JsonFormat

enum class ErrorCode(val status: Int, val code: String, val message: String) {
    // Common
    INVALID_INPUT_VALUE(400, "C001", " Invalid Input Value"),
    METHOD_NOT_ALLOWED(405, "C002", " Invalid Input Value"),
    ENTITY_NOT_FOUND(400, "C003", " Entity Not Found"),
    INTERNAL_SERVER_ERROR(500, "C004", "Server Error"),
    INVALID_TYPE_VALUE(400, "C005", " Invalid Type Value"),
    SERVICE_ERROR(400, "C006", "Access is Denied"),
}